### PR TITLE
Customizer tweaks

### PR DESCRIPTION
**Pull request by mikejolley**
Addresses the issues in #18586 

- WooCommerce Customizer panels visible at all times
- Clicking WC Customizer panel will redirect to shop page.
- Upgrading will calculate thumbnail aspect ratio from old settings rather than forcing 1:1 cropping mode.

### CODE REVIEW

File: includes/wc-core-functions.php [hunk 1]
```diff
@@ -2004,3 +2004,42 @@ function wc_cleanup_session_data() {
...
+ * @return array|bool a 1/2 would be [1, 2] array (this can be imploded with '/' to form a string).
+ */
+function wc_decimal_to_fraction( $decimal ) {
+	if ( 0 > $decimal || ! is_numeric( $decimal ) ) {
+		// Negative digits need to be passed in as positive numbers and prefixed as negative once the response is imploded.
+		return false;
+	}
+
+	if ( 0 === $decimal ) {
+		return [ 0, 0 ];
```
Comment (claudiulodro) [inline includes/wc-core-functions.php]: This syntax isn't supported on old PHP versions (https://travis-ci.org/woocommerce/woocommerce/jobs/333246051)

Comment (claudiulodro) [reply includes/wc-core-functions.php]: And it should be [ 0, 1 ]. [0/0 is undefined](https://brilliant.org/wiki/what-is-0-0/). :P

File: includes/customizer/class-wc-shop-customizer.php [hunk 2]
```diff
@@ -503,15 +501,15 @@ private function add_product_images_section( $wp_customize ) {
...
+		include_once WC_ABSPATH . 'includes/customizer/class-wc-customizer-control-cropping.php';
 
 		$wp_customize->add_setting(
 			'woocommerce_thumbnail_cropping',
 			array(
-				'default'              => '1:1',
-				'type'                 => 'option',
-				'capability'           => 'manage_woocommerce',
-				'sanitize_callback'    => 'wc_clean',
+				'default'           => '1: 1',
```
Comment (claudiulodro) [inline includes/customizer/class-wc-shop-customizer.php]: I don't think that space is supposed to be there.

**Comment by claudiulodro** [comment]
Looks good just a couple small tweaks needed.

**Comment by claudiulodro** [comment]
LGTM. I'll merge soon as CI is finished running.

**Comment by SorinGFS** [comment]
@mikejolley I just dont get it, sorry, looks like we talk about two different things.

Category page has a single image for every product, once you click the product to enter single-page, those little thumbnails are now at same size as category images, so the load will be like in the next picture, I'm wrong?

The cache enter in his role just when an user comes back to the same single-page

<a href="https://ibb.co/gZ5FJb">Untitl333ed</a>

Sorry for posting here, I didn't notice that the issue #18586 was closed

**Comment by SorinGFS** [comment]
@mikejolley I have few sugetions:
1.For this transition:
 a) to make it optional, placing a checkbox exactly where the settings are located now (settings>products>display), something like Responsive Thumbnails Y/N
 b) In the same place, in case user opts for responsive thumbnails two more settings to be added, max thumbnails per row, and thumbnail container max height
 c) In case user opts for responsive thimbnails to put an informative tooltip for user to know where the settings can be modified (in Customizer)
2.For the tumbnail container a setting in pixels for the distance from where is placed first tumbnail, as you can see in picture there is a gap, default should be zero, and in user opts for different number of thumnails per row he willl have the option to center those tumbnails. In fact the number of thumnails per row should be general, no matter what is the user option.

**Comment by mikejolley** [comment]
@SorinGFS You're really going off topic here. Your main gripe (the only one we've not addressed) is that the thumbnails are slightly larger on product pages. I don't see this as a major issue or blocker. If the images *look* larger, thats a theme issue anyway. CSS controls the display size of thumbnails.

**Comment by SorinGFS** [comment]
@mikejolley  Can we control the number of thumbnails per row in css?

**Comment by mikejolley** [comment]
Yes of course. They are set to 25% width by default (4 col).

**Comment by SorinGFS** [comment]
just css wont help I think, a litle php is needed:
`function number_of_columns_product_gallery( $wrapper_classes ) {
    $columns = 5; 
    $wrapper_classes[2] = 'woocommerce-product-gallery--columns-' . absint( $columns );
    return $wrapper_classes;
}`

**Comment by mikejolley** [comment]
Ah there is a filter too `woocommerce_product_thumbnails_columns`. But this is not new in 3.3 nor is it related to image sizes!

**Comment by SorinGFS** [comment]
@mikejolley Ok. Good to know. That wasn't my biggest concern anyway, but the fact that after upgrade all the existing sites will have to solve the transition in the next second, because their websites will look different after upgrade. IMHO a transition like this should be optional. (I mean the display part)

**Pull request merged**